operator/pkg/ciliumenvoyconfig/envoy_config.go

```diff
@@ -0,0 +1,340 @@
+	}
+	return envoyConfig, exists, err
+}
+
+func (em *envoyConfigManager) MarkSynced() {
+	em.informer.HasSynced()
+}
+
+func (m *manager) getEnvoyConfigForService(svc *slim_corev1.Service) (*ciliumv2.CiliumEnvoyConfig, error) {
+	resources, err := m.getResources(svc)
+	if err != nil {
+		return nil, err
+	}
+	return &ciliumv2.CiliumEnvoyConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       ciliumv2.CECKindDefinition,
+			APIVersion: ciliumv2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", ciliumEnvoyLBPrefix, svc.GetName()),
+			Namespace: svc.GetNamespace(),
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: slim_corev1.SchemeGroupVersion.String(),
+					Kind:       "Service",
+					Name:       svc.GetName(),
+					UID:        svc.GetUID(),
+				},
+			},
+		},
+		Spec: ciliumv2.CiliumEnvoyConfigSpec{
+			Services: []*ciliumv2.ServiceListener{
+				{
+					Name:      svc.GetName(),
+					Namespace: svc.GetNamespace(),
+				},
+			},
+			Resources: resources,
+		},
+	}, nil
+}
+
+func (m *manager) getResources(svc *slim_corev1.Service) ([]ciliumv2.XDSResource, error) {
+	var resources []ciliumv2.XDSResource
+	listener, err := m.getListenerResource(svc)
+	if err != nil {
+		return nil, err
+	}
+	resources = append(resources, listener)
+
+	routeConfig, err := m.getRouteConfigurationResource(svc)
+	if err != nil {
+		return nil, err
+	}
+	resources = append(resources, routeConfig)
+
+	clusters, err := m.getClusterResources(svc)
+	if err != nil {
+		return nil, err
+	}
+	resources = append(resources, clusters...)
+	return resources, nil
+}
+
+func (m *manager) getClusterResources(svc *slim_corev1.Service) ([]ciliumv2.XDSResource, error) {
+	lbPolicy, ok := envoy_config_cluster_v3.Cluster_LbPolicy_value[strings.ToUpper(m.algorithm)]
+	if !ok {
+		lbPolicy = int32(envoy_config_cluster_v3.Cluster_ROUND_ROBIN)
+	}
+
+	cluster := &envoy_config_cluster_v3.Cluster{
+		Name:           getName(svc),
+		ConnectTimeout: &durationpb.Duration{Seconds: 5},
+		LbPolicy:       envoy_config_cluster_v3.Cluster_LbPolicy(lbPolicy),
+		TypedExtensionProtocolOptions: map[string]*anypb.Any{
+			"envoy.extensions.upstreams.http.v3.HttpProtocolOptions": toAny(&envoy_config_upstream.HttpProtocolOptions{
+				UpstreamProtocolOptions: &envoy_config_upstream.HttpProtocolOptions_UseDownstreamProtocolConfig{
+					UseDownstreamProtocolConfig: &envoy_config_upstream.HttpProtocolOptions_UseDownstreamHttpConfig{
+						// Empty HTTP/2 options has no effect, so this should not be needed
+						Http2ProtocolOptions: &envoy_config_core_v3.Http2ProtocolOptions{},
+					},
+				},
+			}),
+		},
+		OutlierDetection: &envoy_config_cluster_v3.OutlierDetection{
+			SplitExternalLocalOriginErrors: true,
+			// The number of consecutive locally originated failures before ejection occurs.
+			ConsecutiveLocalOriginFailure: &wrapperspb.UInt32Value{Value: 2},
```

Comment:
The default is 5, why do we pick 2?

Reply:
good point :100:, it's better to just stick with the default right now :100: